### PR TITLE
jasmine: fix `expect()` typing so it works with `strictNullChecks` and nullable argument

### PR DESCRIPTION
**Pull request by mijay**
During #21305 `expect<T>(actual: T)` signature was changed to mark `actual` as optional parameter. This breaks the type inference in cases where passed value is nullable: e.g. when `actual` value is of type `string | undefined`, typescript infers `T` as `string`, and hence `toEqual(undefined)` is not allowed with enabled `strictNullChecks`.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [ ] ~Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>~
- [ ] ~Increase the version number in the header if appropriate.~
- [ ] ~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~

/cc @lukas-zech-software

### CODE REVIEW

**Comment by gmoothart** [comment]
shouldn't you be using `toBeUndefined` ?

**Comment by gmoothart** [comment]
Oh, I understand sorry. The reason for the change is the new `expect().nothing()` construct explained in https://github.com/jasmine/jasmine/issues/1221

So we need to support both.

**Comment by mijay** [comment]
Done. Apparently separate override for `expect()` works.

**Review by lukas-zech-software — CHANGES_REQUESTED** [review]
I would propose to extract the new `nothing` matcher to an own `NothingMatcher` interface

```typescript
   declare function expect(): jasmine.NothingMatchers;
   
   interface NothingMatchers {
        nothing(): Any;
    }

```

this way `nothing` can only be called if no parameter is provided to `expect` and cannot be called if one was provided

**Comment by mijay** [comment]
Done!

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -88,7 +88,7 @@ declare function expect<T>(actual: ArrayLike<T>): jasmine.ArrayLikeMatchers<T>;
  * Create an expectation for a spec.
  * @param actual Actual computed value to test expectations against.
  */
-declare function expect<T>(actual?: T): jasmine.Matchers<T>;
+declare function expect<T>(actual: T): jasmine.Matchers<T>;
```
[the remark this answers is not in the document]
Comment (mijay) [reply types/jasmine/index.d.ts]: done

**Comment by lukas-zech-software** [comment]
nice! :+1:

File: types/jasmine/jasmine-tests.ts [hunk 2]
```diff
@@ -54,6 +54,11 @@ describe("Included matchers:", () => {
             };
             expect(foo).toEqual(bar);
         });
+
+        it("should work for optional values", () => {
+            var opt: string | undefined = "s";
+            expect(opt as (string | undefined)).toEqual(undefined);
```
Comment (gmoothart) [inline types/jasmine/jasmine-tests.ts]: the cast here seems redundant?

**Review by gmoothart — APPROVED** [review]

**Pull request merged**